src/verifier/tests.rs

```diff
@@ -19,6 +19,58 @@ pub fn setup_logger() {
 const VALID_SCRIPTS: &[&str] =
     &["wasm:bytecode:call:alt { new_target_fn_name = redirect_to_fault_injector; }"];
 
+const TYPE_ERROR_SCRIPTS: &[&str] = &[
+    // predicate
+    // note that this will have cascading type check errors
+    // might want to make type check errors fatal so that we can stop early
```

Comment:
The worry is that the expression `((1 == "str") && true && .... && true)` will cause type error for each `&&` since it always can't get the lhs. The following struct (as we discussed for return statements) might fix the problem.
```rust
enum OptType {
  Some(DataType),
  AssumeGood,
  None
}
```